### PR TITLE
#18037 Fix forceSingleConnection set for single connection drivers

### LINKED ISSUE #18037: since latest update I'm getting many more key-pair authentication attempts per session
### Description

1. connect to Snowflake
2. receive push notification on phone to approve key-pair authentication (this was normal behavior prior to the latest update, but since then:)
3. receive 2nd push notification on phone to approve key-pair authentication (immediately after approving first one)
4. receive another push notification for each script I open in connection to the same Snowflake session

### DBeaver Version

Version 22.2.3.202210231354

### Operating System

macOS monterey 12.6

### Database and driver

Database name and version 6.18.2
Driver name Snowflake

### Steps to reproduce

1. connect to Snowflake
2. receive push notification on phone to approve key-pair authentication (this was normal behavior prior to the latest update, but since then:)
3. receive 2nd push notification on phone to approve key-pair authentication (immediately after approving first one)
4. receive another push notification for each script I open in connection to the same Snowflake session

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by E1izabeth**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java [hunk 1]
```diff
@@ -1087,6 +1087,10 @@ private boolean askForSSHJumpServerPassword(@NotNull DBWHandlerConfiguration tun
     }
 
     public void openDataSource(DBRProgressMonitor monitor, boolean initialize) throws DBException {
+        if (this.getDriver().isSingleConnection()) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java]: Maybe its better to set default value of `forceUseSingleConnection` in constructor?

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java [hunk 2]
```diff
@@ -322,6 +325,9 @@ DBPDataSourceOrigin getOriginSource() {
 
     public void setDriver(@NotNull DriverDescriptor driver) {
         this.driver = driver;
+        if (this.driver.isSingleConnection()) {
+            this.setForceUseSingleConnection(true);
+        }
```
Comment (Destrolaric) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java]: Not sure about adding this kind of interaction inside setter

Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java]: I'd prefer to replace it with 
`this.forceUseSingleConnection = driver.isSingleConnection();`

**Review by Destrolaric — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**